tools/test.py

```diff
@@ -173,7 +173,10 @@ def main():
         if args.eval:
             eval_kwargs = cfg.get('evaluation', {}).copy()
             # hard-code way to remove EvalHook args
-            for key in ['interval', 'tmpdir', 'start', 'gpu_collect']:
+            for key in [
+                    'interval', 'tmpdir', 'start', 'gpu_collect', 'save_best',
+                    'rule', 'by_epoch'
+            ]:
```

Comment:
Too ugly here. we can modify as following:
`eval_hook_args = [ 'interval', 'tmpdir', 'start', 'gpu_collect', 'save_best' , 'rule', 'by_epoch']`
`for key in eval_hook_args:`